libraries/ESP8266WiFi/src/WiFiServerBearSSL.cpp

```diff
@@ -0,0 +1,91 @@
+                   with Arduino WiFi shield library
+
+  Copyright (c) 2018 Earle F. Philhower, III
+
+  This library is free software; you can redistribute it and/or
+  modify it under the terms of the GNU Lesser General Public
+  License as published by the Free Software Foundation; either
+  version 2.1 of the License, or (at your option) any later version.
+
+  This library is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+  Lesser General Public License for more details.
+
+  You should have received a copy of the GNU Lesser General Public
+  License along with this library; if not, write to the Free Software
+  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+*/
+
+#define LWIP_INTERNAL
+
+extern "C" {
+    #include "osapi.h"
+    #include "ets_sys.h"
+}
+
+#include "debug.h"
+#include "ESP8266WiFi.h"
+#include "WiFiClient.h"
+#include "WiFiServer.h"
+#include "lwip/opt.h"
+#include "lwip/tcp.h"
+#include "lwip/inet.h"
+#include "include/ClientContext.h"
+#include "WiFiServerBearSSL.h"
+
+// Only need to call the standard server constructor
+WiFiServerBearSSL::WiFiServerBearSSL(IPAddress addr, uint16_t port) : WiFiServer(addr, port)
+{
+}
+
+// Only need to call the standard server constructor
+WiFiServerBearSSL::WiFiServerBearSSL(uint16_t port) : WiFiServer(port)
+{
+}
+
+// Specify a RSA-signed certificate and key for the server.  Only copies the pointer, the
+// caller needs to preserve this chain and key for the life of the object.
+void WiFiServerBearSSL::setRSACert(const BearSSLX509List *chain, const BearSSLPrivateKey *sk)
+{
+  _chain = chain;
+  _sk = sk;
+}
+
+// Specify a EC-signed certificate and key for the server.  Only copies the pointer, the
+// caller needs to preserve this chain and key for the life of the object.
+void WiFiServerBearSSL::setECCert(const BearSSLX509List *chain, unsigned cert_issuer_key_type, const BearSSLPrivateKey *sk)
+{
+  _chain = chain;
+  _cert_issuer_key_type = cert_issuer_key_type;
+  _sk = sk;
+}
+
+// Return a client if there's an available connection waiting.  If one is returned,
+// then any validation (i.e. client cert checking) will have succeeded.
+WiFiClientBearSSL WiFiServerBearSSL::available(uint8_t* status)
+{
+  (void) status; // Unused
+  if (_unclaimed) {
+    if (_sk && _sk->isRSA()) {
+      WiFiClientBearSSL result(_unclaimed, _chain, _sk, _iobuf_in_size, _iobuf_out_size, _client_CA_ta);
+      _unclaimed = _unclaimed->next();
+      result.setNoDelay(_noDelay);
+      DEBUGV("WS:av\r\n");
+      return result;
+    } else if (_sk && _sk->isEC()) {
+      WiFiClientBearSSL result(_unclaimed, _chain, _cert_issuer_key_type, _sk, _iobuf_in_size, _iobuf_out_size, _client_CA_ta);
+      _unclaimed = _unclaimed->next();
+      result.setNoDelay(_noDelay);
+      DEBUGV("WS:av\r\n");
+      return result;
+    }
```

Comment:
else what? (error handling)

Reply:
By default the fallthrough to returning a non-attached Client is the best that can be done.  I'll add in a DEBUGV message, though.  I don't want to actually do an accept() (i.e. move it from _unused queue) as then I think you'll actually get a TCP connect back on the client who's trying, which is not what we want...the server can't handle the connection, so let them sit unanswered.

Reply:
Maybe it's late still out there, but the function already ends with a return (unconnectedClient). :)  I'll add a comment, though!